types/src/validator_verifier.rs

```diff
@@ -232,33 +233,38 @@ impl ValidatorVerifier {
         message: &T,
         multi_signature: &MultiSignature,
     ) -> std::result::Result<(), VerifyError> {
+        // Verify the multi signature is not empty
+        multi_signature
+            .multi_sig()
+            .as_ref()
+            .ok_or(VerifyError::InvalidSignature)?;
+        // Verify the number of signature is not greater than expected.
         self.check_num_of_signatures(multi_signature)?;
-        self.check_voting_power(
-            multi_signature
-                .get_voter_addresses(&self.get_ordered_account_addresses_iter().collect_vec())
-                .iter(),
-        )?;
+        let authors: HashSet<AccountAddress> = multi_signature
```

Comment:
I was expecting a Vec\<AccountAddress\> instead of a HashSet
```
authors
  .iter()
  .filter_map(|author| 
    self.address_to_validator_info
      .get(author)
      .map(|info| info.consensus_key)
  )
```

Reply:
Good point - didn't realize `address_to_validator_info` already is a BTreeMap.